homeassistant/components/livisi/switch.py

```diff
@@ -0,0 +1,166 @@
+"""Code to handle a Livisi switches."""
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.components.switch import SwitchEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import (
+    DOMAIN,
+    ID,
+    IS_REACHABLE,
+    LIVISI_REACHABILITY_CHANGE,
+    LIVISI_STATE_CHANGE,
+    LOGGER,
+    PSS_DEVICE_TYPE,
+    STATE,
+)
+from .coordinator import LivisiDataUpdateCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up switch device."""
+    coordinator: LivisiDataUpdateCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+    @callback
+    def handle_coordinator_update() -> None:
+        """Add switch."""
+        shc_devices: list[dict[str, Any]] = coordinator.data
+        entities: list[SwitchEntity] = []
+        for device in shc_devices:
+            if device["type"] == PSS_DEVICE_TYPE:
+                if device["id"] not in coordinator.devices:
```

Comment:
Please combine the two checks with `and`.

Reply:
I changed this.